qt5 --with-mysql fails to build on macOS: 10.13.5-x86_64

**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [x ] are reporting a bug others will be able to reproduce and not asking a question. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh
- [x ] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, official formula (not cask)? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x ] ran `brew update` and can still reproduce the problem?
- [x ] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x ] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?

https://gist.github.com/apaiva/f56698fc9eff0ba407829186d42a56f1

- [x ] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

brew gist-logs worked.

To help us debug your issue please explain:
- What you were trying to do (and why)

Reinstall qt5 with mysql support.   I've been using qt5 via pyqt for some time now and recently needed to add mysql support.  I've uninstalled qt5 and attempted to install again with "brew install -v qt5 --with-mysql"; however, the build fails.

- What happened (include command output)

Full output is in the gist, but here's the first error is:

qsql_mysql.cpp:235:9: error: unknown type name 'my_bool'
        my_bool nullIndicator;
        ^

- What you expected to happen

Clean build.

- Step-by-step reproduction instructions (by running `brew install` commands)

brew install -v --build-from-source qt5 --with-mysql

--- a/Formula/qt.rb
+++ b/Formula/qt.rb
@@ -21,9 +21,11 @@ class Qt < Formula
   option "with-examples", "Build examples"
   option "without-proprietary-codecs", "Don't build with proprietary codecs (e.g. mp3)"
 
+  deprecated_option "with-mysql" => "with-mysql-client"
+
   depends_on "pkg-config" => :build
   depends_on :xcode => :build
-  depends_on "mysql" => :optional
+  depends_on "mysql-client" => :optional
   depends_on "postgresql" => :optional
 
   # Restore `.pc` files for framework-based build of Qt 5 on macOS, partially
@@ -56,7 +58,7 @@ class Qt < Formula
 
     args << "-nomake" << "examples" if build.without? "examples"
 
-    if build.with? "mysql"
+    if build.with? "mysql-client"
       args << "-plugin-sql-mysql"
       (buildpath/"brew_shim/mysql_config").write <<~EOS
         #!/bin/sh

